packages/telemetry/src/appInsights.ts

```diff
@@ -0,0 +1,206 @@
+import { Logger, Messages } from '@salesforce/core';
+import { AsyncCreatable, Env } from '@salesforce/kit';
+import { isBoolean, isNumber, isString, JsonPrimitive } from '@salesforce/ts-types';
+import * as appInsights from 'applicationinsights';
+import * as os from 'os';
+
+export { TelemetryClient } from 'applicationinsights';
+
+export type Properties = {
+  [key: string]: string;
+};
+
+export type Measurements = {
+  [key: string]: number;
+};
+
+export type Attributes = {
+  [key: string]: JsonPrimitive | undefined;
+};
+
+export interface TelemetryOptions {
+  project: string;
+  key: string;
+  commonProperties?: Properties;
+  contextTags?: Properties;
+  env?: Env;
+  gdprSensitiveKeys?: string[];
+}
+
+Messages.importMessagesDirectory(__dirname);
+
+/**
+ * This is a wrapper around appinsights sdk for convenience.
+ *
+ * NOTE: THis should not be used directly. Use TelemetryReporter which
+ * will check if telemetry is disabled and do GDPR checks.
+ */
+export class AppInsights extends AsyncCreatable<TelemetryOptions> {
+  public static GDPR_HIDDEN = '<GDPR_HIDDEN>';
+  private static ASIMOV_ENDPOINT = 'https://vortex.data.microsoft.com/collect/v1';
+  public appInsightsClient!: appInsights.TelemetryClient;
+  private options: TelemetryOptions;
+  private logger!: Logger;
+  private env!: Env;
+  private gdprSensitiveKeys: string[] = [];
+
+  constructor(options: TelemetryOptions) {
```

Comment:
Shouldn't there be a check here or on `init()` to verify the user has not opted out from sending telemetry ?

Reply:
I have it in the `TelemetryReporter`. It seemed weird having it in both places and a total duplicate of logic. I have a note that people should only be instantiating `TelemetryReporter` and that is the only one exposed too.

Reply:
@amphro sounds good 👍